### PR TITLE
fix user dropdown and entity restrict

### PR DESCRIPTION
**Pull request by orthagh**
- add a defaut value for entity_restrict
- striplashes before json_decode on this parameter

see #1242
see #1086

### CODE REVIEW

**Comment by MicaENSL** [comment]
Bonjour,
je me permets de poster ici car j'ai patché le fichier getDropdownUsers.php avec le dernier commit. Quand je cherche un ticket par demandeur j'ai bien toute la liste des utilisateurs GLPI, par contre quand je fais une recherche "Attribué à -> Technicien" la liste déroulante n'est plus limitée aux techniciens de l'entité mais renvoie tous les utilisateurs.
Merci d'avance de votre réponse.

MC

File: ajax/getDropdownUsers.php [hunk 1]
```diff
@@ -77,10 +77,12 @@
...
 }
 
+$entity_restrict = -1;
 if (isset($_POST['entity_restrict'])) {
-   $entity_restrict = json_decode($_POST['entity_restrict']);
+   $entity_restrict = Toolbox::stripslashes_deep($_POST['entity_restrict']);
+   $entity_restrict = json_decode($entity_restrict);
    if (json_last_error() != JSON_ERROR_NONE) {
-      $entity_restrict = $_POST['entity_restrict'];
+      $entity_restrict = $entity_restrict;
```
Comment (trasher) [inline ajax/getDropdownUsers.php]: Sounds lke there is something wrong here...

**Comment by trasher** [comment]
I've commited a attempt to fix that. As I'm in trouble to generate required data to make such a test, would you please test if the latest commit solves the issue?

Thank you!

**Comment by MicaENSL** [comment]
Sorry, after applying the patch the first issue happens again: when I perform a search with "Demandeur" or "Attribué à", the dropdown list shows only users in "Entité racine".

**Comment by trasher** [comment]
OK, thank you @MicaENSL ; I'll try to populate a local database to reproduce the issue.

**Comment by trasher** [comment]
@MicaENSL could you try again please?

**Comment by PierreTeclib** [comment]
Hello, It's work fine for me!

Thks

**Comment by trasher** [comment]
@PierreTeclib thank you!

**Comment by MicaENSL** [comment]
Sorry, when I search "Attribué à - technicien" it still display all users and not only technicians.

**Comment by trasher** [comment]
OK, I think I've finally find out what was wrong. Entity issue is one thing, but we had another issue (explained in #1242 ).
This tie, I hope problem is finally solved; if you can again take a look @MicaENSL (I still dot not have reliable data unfortunately).

**Comment by MicaENSL** [comment]
Great! It works fine now, thank you.

**Comment by trasher** [comment]
@MicaENSL great news, thanks for your patience :)

**Pull request merged**